### PR TITLE
added HashTable

### PR DESCRIPTION
**Pull request by MaxGmn**

### CODE REVIEW

File: ArrayDictionarySet/ArrayDictionarySet/ViewController.swift [hunk 1]
```diff
@@ -440,6 +462,78 @@ class MySet<V:Comparable>: CustomStringConvertible{
...
+        guard let neededItem = itemsArray[neededItemsHash] else{
+            throw MyDictionaryErrors.itemIsNotExist
+        }
+        
+        return neededItem.value
+    }
+    
+    func getHashCode(key:K) -> Int{
+        let hashValue = abs(key.hashValue)
+        if itemsArray.count != 0 && hashValue > itemsArray.count{
```
Comment (ghost) [inline ArrayDictionarySet/ArrayDictionarySet/ViewController.swift]: this check is useless you can always return "return hashValue % itemsArray.count"

File: ArrayDictionarySet/ArrayDictionarySet/ViewController.swift [hunk 2]
```diff
@@ -440,6 +462,73 @@ class MySet<V:Comparable>: CustomStringConvertible{
...
+    func add(key:K, value:V){
+        
+        let newItemsHash = getHashCode(key: key)
+        let newNode = HashDictionaryNode(key: key, value: value, hash: newItemsHash)
+        
+        itemsArray[newItemsHash] = newNode
+    }
+    
+    func remove(key:K) throws{
+        let remoweItemsHash = getHashCode(key: key)
```
Comment (oleksii-lubianyi-epam) [inline ArrayDictionarySet/ArrayDictionarySet/ViewController.swift]: `remoVe`

File: ArrayDictionarySet/ArrayDictionarySet/ViewController.swift [hunk 3]
```diff
@@ -440,6 +462,73 @@ class MySet<V:Comparable>: CustomStringConvertible{
...
+        
+        guard let _ = itemsArray[remoweItemsHash] else{
+            throw MyDictionaryErrors.itemIsNotExist
+        }
+        
+        itemsArray.remove(at: remoweItemsHash)
+    }
+    
+    func getItem(key:K) throws-> V{
+        let neededItemsHash = getHashCode(key: key)
```
Comment (oleksii-lubianyi-epam) [inline ArrayDictionarySet/ArrayDictionarySet/ViewController.swift]: `needed` -> `requested`

**Review by oleksii-lubianyi-epam — APPROVED** [review]

**Comment by oleksii-lubianyi-epam** [comment]
Please resolve conflicts

**Pull request merged**